### DISCUSSION TITLE
react native storage upload File

### DISCUSSION THREAD

**Discussion by megacherry** [supabase/supabase#2106] [Questions]
hi everybody,

can someone help me how i can upload a image in react native to supabase. i have googled around but i found no easy or handy solution. i know i have to convert the image uri to blob/file. i wanted to use this library: react-native-blob-util and looked the examples https://github.com/RonRadtke/react-native-blob-util/blob/master/examples/ReactNativeBlobUtil/App.js

i can upload the image, but it has 0 byte size.

thanks in advance.

regards
Andreas

**Comment by fenos** [comment]
can you please provide a code snippet of your upload please? I might be able to spot the issue.
Usually when the file is 0 bytes is because we don't wait for the promise that reads the file in memory

**Reply by megacherry** [reply]
> can you please provide a code snippet of your upload please? I might be able to spot the issue. Usually when the file is 0 bytes is because we don't wait for the promise that reads the file in memory

Thank you for asking. I have solved the problem in the meantime. If anyone stumbles across a similar problem, they can leave a quick comment here and I'll add sample code.

**Reply by fefox74** [reply]
hey can you help me, im trying to save the image by transforming a base64 string that i got from a canvas to an array buffer but its not working, its weird, because this is working when i get the base64 of an image using expo image picker

**Reply by Myzel394** [reply]
Hey @megacherry why not share your code directly? @fefox74 Did you find a solution?

**Reply by fefox74** [reply]
yes, i found a solution, i saved the base64 in the file system and then i got the base 64 again and it worked
